tf2/src/buffer_core.cpp

```diff
@@ -528,23 +528,23 @@ tf2::TF2Error BufferCore::walkToTopParent(F& f, TimePoint time, CompactFrameID t
   if (frame_chain)
   {
     // Pruning: Compare the chains starting at the parent (end) until they differ
-    int m = (int)reverse_frame_chain.size()-1;
-    int n = (int)frame_chain->size()-1;
-    for (; m >= 0 && n >= 0; --m, --n)
+    size_t fcs = frame_chain->size() - 1;
+    size_t rcs = reverse_frame_chain.size() - 1;
+    size_t m = 1;
+    size_t n = 1;
+    for (; m <= rcs && n <= fcs; ++m, ++n)
     {
-      if ((*frame_chain)[n] != reverse_frame_chain[m])
+      if ((*frame_chain)[fcs - n] != reverse_frame_chain[rcs - m])
         break;
     }
     // Erase all duplicate items from frame_chain
-    if (n > 0)
-      frame_chain->erase(frame_chain->begin() + (n-1), frame_chain->end());
+    if (n <= fcs) {
```

Comment:
@ivanpauno I think it should be
```suggestion
    if (n < fcs) {
```
instead.

Reply:
done